types/comparable_bytes.cc

```diff
@@ -1231,6 +1231,11 @@ struct to_comparable_bytes_visitor {
 
     void operator()(const empty_type_impl&) {}
 
+    void operator()(const date_type_impl&) {
+        out.write(serialized_bytes_view.prefix(sizeof(db_clock::rep)));
+        serialized_bytes_view.remove_prefix(sizeof(db_clock::rep));
+    }
+
```

Comment:
@lkshminarayanan please review.

Don't we need tests for Cassandra compatibility?

Reply:
> I assumed date_type_impl was not in use anymore, so I had skipped it.

It can't be created in new tables (created via CQL) anymore, but it still can be present in (very) old tables, and tests still generate it.